src/ng-select/lib/ng-select.component.ts

```diff
@@ -376,7 +373,8 @@ export class NgSelectComponent implements OnDestroy, OnChanges, OnInit, AfterVie
 	handleKeyCode($event: KeyboardEvent) {
 		const target = $event.target;
 
-		if (this.clearButton && this.clearButton.nativeElement === target) {
+		const clearButton = this.clearButton();
```

Comment:
Nit: I don't we need separate variable here

Reply:
Updated - https://github.com/ng-select/ng-select/pull/2584/commits/53cf59e5182d0cbc93ac9608a82ee5b0dc0c44be